04_numeronym/shahriys/main.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+var outnum io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	var returnval = make([]string, 0, len(vals))
+	for _, val := range vals {
+		switch len(val) {
+		case 0, 1, 2, 3:
+			returnval = append(returnval, val)
+		default:
+			l := len(val) - 2
+			s := string(val[0]) + strconv.FormatInt(int64(l), 10) + string(val[l+1])
```

Comment:
An alternate approach which I find a little neater is to use the `fmt.Sprintf()` function.

```go
fmt.Sprintf("%c%d%c", val[0], l, val[l+1])
```

Your's is fine though